src/core_plugins/elasticsearch/lib/cluster.js

```diff
@@ -0,0 +1,133 @@
+import elasticsearch from 'elasticsearch';
+import { get, set, isEmpty, cloneDeep, pick } from 'lodash';
+import toPath from 'lodash/internal/toPath';
+import Boom from 'boom';
+
+import filterHeaders from './filter_headers';
+import { parseConfig } from './parse_config';
+
+export default class Cluster {
+  constructor(config) {
+    this._config = Object.assign({}, config);
+    this.errors = elasticsearch.errors;
+
+    this._client = this.createClient();
+    this._noAuthClient = this.createClient({ auth: false });
+
+    return this;
+  }
+
+  /**
+   * callWithRequest
+   *
+   * Performs a call to ES, passing through whitelisted headers in the request
+   *
+   * @param {Object|undefined} req - The request object
+   * @param {string} endpoint
+   * @param {Object} clientParams
+   * @param {Object} options
+   * @param {boolean} options.wrap401Errors
+   * @returns {Promise}
+   */
+  callWithRequest = (req = {}, endpoint, clientParams = {}, options = {}) => {
```

Comment:
Yes, this will work if req is undefined and endpoint is passed in. req would be an empty object.

This isn't doing reassignment, simply ensuring req is an object if passed undefined.